sequencing/clinical_annotator/resources/home/dnanexus/ClinVar_tsvTOvcf.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+"""Script to reformat the ClinVar TSV file into two VCF files, one for b37 and
+one for b38.
+"""
+__author__ = "Thomas Nate Person"
+__license__ = "GPLv3"
+__email__ = "thomas.n.person@gmail.com"
+
+import re
+import sys
+import time
+import gzip
+
+MonthYear =time.strftime("%b")+time.strftime("%Y")
+DayMonthYear =time.strftime("%d")+time.strftime("%b")+time.strftime("%Y")
+
+dbFile = gzip.open(sys.argv[1],'r')
+db_b37_vcf = open(sys.argv[1].replace("txt.gz","")+'b37.vcf','w')
+db_b38_vcf = open(sys.argv[1].replace("txt.gz","")+'b38.vcf','w')
+
+
+for line in dbFile:
+	if line.startswith("#AlleleID"):
+		headerFields="|".join(line.lstrip("#").strip().split("\t"))
+		headerFields="Allele|"+headerFields
+	else:
+		break
+
+db_b37_vcf.write("##fileformat=VCFv4.2\n")
+db_b37_vcf.write("##INFO=<ID=ClinVar.TSV."+DayMonthYear+",Number=.,Type=String,Description=\"ClinVar.TSV." + DayMonthYear+": '"+headerFields+"'\">\n")
+db_b37_vcf.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n")
+
+db_b38_vcf.write("##fileformat=VCFv4.2\n")
+db_b38_vcf.write("##INFO=<ID=ClinVar.TSV."+DayMonthYear+",Number=.,Type=String,Description=\"ClinVar.TSV." + DayMonthYear+": '"+headerFields+"'\">\n")
+db_b38_vcf.write("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\n")
+
+CHROM =-1
+POS = -1
+REF = -1
+ALT = -1
+ASSEMBLY = -1
+
+dbFile.seek(0)
+
+new_vcf=dict()
+new_vcf_b37=dict()
+new_vcf_b38=dict()
+new_vcf["GRCh37"]=new_vcf_b37
+new_vcf["GRCh38"]=new_vcf_b38
+new_vcf["source"]="ClinVar.TSV."+DayMonthYear+"="
+
+for line in dbFile:
+	if line.startswith("#"):
```

Comment:
do header = dbFile.next() and then do this block before the for loop. Not necessary to check if the line starts with '#' if you know the first line will always be the header. Then you can do all the preprocessing in a single pass (without having to seek back to the start of the file and do a second pass).